lib/membrane_rtc_engine/engine.ex

```diff
@@ -1248,12 +1259,15 @@ defmodule Membrane.RTC.Engine do
     end)
   end
 
-  defp build_raw_format_links(subscriptions, ctx, state) do
+  defp build_raw_format_links(subscriptions, ctx, state, prepared_raw_format_tees) do
     subscriptions
     |> Stream.filter(&(&1.format == :raw))
     |> Stream.map(& &1.track_id)
     |> Stream.uniq()
-    |> Stream.reject(&Map.has_key?(ctx.children, {:raw_format_tee, &1}))
+    |> Stream.reject(fn track_id ->
+      raw_tee = {:raw_format_tee, track_id}
+      Map.has_key?(ctx.children, raw_tee) or raw_tee in prepared_raw_format_tees
+    end)
     |> Enum.map(&build_raw_format_link(&1, state))
```

Comment:
```suggestion
    subscriptions
    |> Stream.filter(&(&1.format == :raw))
    |> Stream.map(& Map.get(state.tracks, &1.track_id))
    |> Stream.uniq_by(& &1.id)
    |> Stream.reject(&Map.has_key?(ctx.children, {:raw_format_tee, &1.id}) or &1.format == [:raw])
    |> Enum.map(&build_raw_format_link(&1, state))
```

The question is how to get track from state as at the moment there isn't something like `state.tracks`.

Reply:
We can go through all endpoints or add `state.tracks`.

Reply:
Great, so maybe let's use `get_track`. This is not the most efficient way but the code will be simpler. We don't also do this very often